proxy/src/http/source.rs

```diff
@@ -95,12 +79,15 @@ where
 }
 
 /// `GET /branches/<project_id>`
-fn branches_filter(
-    peer: Arc<Mutex<coco::PeerApi>>,
-) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+fn branches_filter<R>(
+    ctx: http::Ctx<R>,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone
+where
+    R: registry::Client + 'static,
+{
     path("branches")
         .and(warp::get())
-        .and(http::with_peer(peer))
+        .and(http::with_context(ctx))
```

Comment:
I wonder if it would make sense to still have something like `http::with_peer_api` that passes down the `Api` to the handler functions. That way they don't take any extra data they don't need. Would that be possible?

Reply:
It is possible, but I don't think it's worth the effort at the moment.

Reply:
Word :ok_hand: